roachtest: update version map and fixtures

celiala opened the pull request:
This commit adds the recently released 19.2.11, and 20.1.7
to the version map in PredecessorVersion.

Release note: None (testing change)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55445)
<!-- Reviewable:end -->

celiala commented:
For release tracking issues:

* 19.2.11 - https://github.com/cockroachdb/cockroach/issues/55256
* 20.1.7 - https://github.com/cockroachdb/cockroach/issues/55257

jlinder commented on pkg/cmd/roachtest/test_runner.go:
```diff
@@ -1171,8 +1171,8 @@ func PredecessorVersion(buildVersion version.Version) (string, error) {
 	// (see runVersionUpgrade). The same is true for adding a new key to this
 	// map.
```
The fixtures also need to be updated like in [this PR](https://github.com/cockroachdb/cockroach/pull/50770).

Note, there were some issues with this in the last couple releases that should be fixed now. If you need it, see https://github.com/cockroachdb/cockroach/pull/55253 for reference.

celiala committed:
roachtest: update version map and fixtures

This commit adds the recently released 19.2.11, and 20.1.7
to the version map in PredecessorVersion.

Release note: None (testing change)

celiala commented:
Thanks @arulajmani and @irfansharif to helping me figure out how to generate the fixtures! 💯 

This is RFAL, thank you!

irfansharif commented on pkg/cmd/roachtest/test_runner.go:
```diff
@@ -1171,8 +1171,8 @@ func PredecessorVersion(buildVersion version.Version) (string, error) {
-		"20.2": "20.1.6",
-		"20.1": "19.2.9",
+		"20.2": "20.1.7",
+		"20.1": "19.2.11",
```
I only see the fixtures in `pkg/cmd/roachtest/fixtures/*/checkpoint-v20.1.tgz` being updated, but it seems we're also bumping the version map for 19.2.11? If I've understood correctly, we'll need to generate fixtures for that too. We can do so by changing `newV` below for one run and copying those artifacts in.

celiala replied:
Ah, thank you! Done :)

irfansharif commented on pkg/cmd/roachtest/versionupgrade.go:
```diff
@@ -94,7 +94,7 @@ func runVersionUpgrade(ctx context.Context, t *test, c *cluster, buildVersion ve
 		// can download it from the homepage); if that is not the case use the
 		// empty string which uses the local cockroach binary.
-		newV := "20.1.6"
+		newV := "20.1.7"
```
Here.

celiala commented:
Thanks for the reviews and all the help!

celiala commented:
bors r+
